app/descriptor/views.py

```diff
@@ -243,48 +242,34 @@ def remove_option_value(desc_id, option_index):
 
     # If no resources are affected, just remove the option value.
     if len(option_assocs) == 0:
-        # Index starting from 1 to skip 'Remove this descriptor'
-        remove_value_from_db(descriptor, choice_names[1:], old_value)
+        remove_value_from_db(descriptor, choice_names, old_value)
         return redirect(url_for('descriptor.descriptor_info', desc_id=desc_id))
 
-    # Create the select field for each resource.
-    for oa in option_assocs:
-        setattr(FixAllResourceOptionValueForm, oa.resource.name,
-                SelectField('', coerce=int, choices=choices))
     form = FixAllResourceOptionValueForm()
 
     # Delete the dynamic fields after the form is instantiated
-    for oa in option_assocs:
-        delattr(FixAllResourceOptionValueForm, oa.resource.name)
-
     if form.validate_on_submit():
         for oa in option_assocs:
-            choice = form[oa.resource.name].data
             # Case for 'Remove this descriptor'
```

Comment:
the comments you're making are a different version of the file than i have locally! will re-push